mlflow/recipes/steps/automl/flaml.py

```diff
@@ -37,14 +44,18 @@ def get_estimator_and_best_params(
 
 
 def _create_custom_metric_flaml(
-    metric_name: str, coeff: int, eval_metric: EvaluationMetric
+    task: str, metric_name: str, coeff: int, eval_metric: EvaluationMetric
 ) -> callable:
     def calc_metric(X, y, estimator) -> Dict[str, float]:
         y_pred = estimator.predict(X)
-        builtin_metrics = _get_regressor_metrics(y, y_pred, sample_weights=None)
+        builtin_metrics = (
+            _get_regressor_metrics(y, y_pred, sample_weights=None)
+            if task == "regression"
+            else _get_binary_classifier_metrics(y_true=y, y_pred=y_pred)
+        )
         res_df = pd.DataFrame()
         res_df["prediction"] = y_pred
-        res_df["target"] = y.values
+        res_df["target"] = y
```

Comment:
This is a bug fix. `y` is already a numpy array

Reply:
Why doesn't this come up in regression & is there a test case that we need to add for this workflow?

Reply:
Looks like our current test coverage does not include a custom metric for automl. I'll add one.

https://github.com/mlflow/mlflow/blob/902c103b64a8e3c48ed081aab09c46937a4d3389/tests/recipes/test_train_step.py#L242-L246

Reply:
EDIT: I was wrong. For regression automl we do test for a custom error function.
https://github.com/mlflow/mlflow/blob/902c103b64a8e3c48ed081aab09c46937a4d3389/tests/recipes/test_train_step.py#L517-L525